server/plugin.go

```diff
@@ -490,3 +540,28 @@ func addFields(fields []*model.SlackAttachmentField, title, msg string, short bo
 		Short: model.SlackCompatibleBool(short),
 	})
 }
+
+func (p *Plugin) messageToJSON(message string) (bool, []byte) {
+	messagefields := strings.Split(message, "\n")
+	if len(messagefields) > 0 {
+		//split each line of the cloudformation event message to field and value
+		var fields = make(map[string]string)
+		for _, field := range messagefields[:len(messagefields)-1] {
+			parts := strings.Split(field, "=")
+			if len(parts) == 2 {
+				fields[parts[0]] = parts[1]
+			} else {
+				return false, nil
+			}
+		}
+
+		jsonmessage, err := json.Marshal(fields)
+		if err != nil {
+			p.API.LogError("Error marshaling in messageToJSON", "err", err.Error())
+			return false, nil
+		}
+
+		return true, jsonmessage
+	}
+	return false, nil
+}
```

Comment:
Instead of returning a boolean in this function, we could instead return an error if there is an issue. Something like this for the three lines that are currently returning `false` here:

- format of message fields is incorrect
- error marshaling message into JSON
- no message fields present in message string

Then the caller of this function would log and handle the error. What do you think @ioandim?

Reply:
The function signature would then become:

```go
func messageToJSON(message string) ([]byte, error) {
```

Reply:
The errors with the messages “error marshaling message into JSON” and “no message fields present in message string” can be added without affecting the current implementation . 

However, I am concerned about the error for incorrect format, as it will produce errors in the logs for any type of messages with a different format from the messages of Cloudformation events. As a result, errors will be included in the logs for valid notifications as well, like messages for Cloudwatch alarms whose format is JSON.

Reply:
> However, I am concerned about the error for incorrect format, as it will produce errors in the logs for any type of messages with a different format from the messages of Cloudformation events. As a result, errors will be included in the logs for valid notifications as well, like messages for Cloudwatch alarms whose format is JSON.

If we can determine "this is not a CloudFormation event" versus "this is an invalid CloudFormation event, or there was an error processing a valid CloudFormation event", then maybe we can return a nil JSON from the `messageToJSON` to signal that it should be skipped due to not being a CloudFormation event. If this isn't feasible, then we can return to the solution of using booleans instead of errors.